url.bs

```diff
@@ -2476,39 +2476,83 @@ background information. [[!HTML]]
+
+ <li><p>When the full <a for=url>host</a> cannot be rendered, elide domain labels starting from the
+ front. (Note that bidirectional text means that the front of the string may not appear at the
+ left.)
 </ul>
 
-<p>For the purposes of bidirectional text it should be rendered as if it were in a
-left-to-right embedding. [[!BIDI]]
+<h4 id=url-rendering-i18n>Internationalization and special characters</h4>
+
+<p>International domain names (IDNs), special characters, and bidirectional text should be handled
+with care to prevent spoofing:
+
+<ul class=brief>
+ <li><p>A <a for=/>URL</a>'s <a for=url>host</a> should be rendered using <a>domain to
+ Unicode</a>.
 
-<p class="note no-backref">Unfortunately, as rendered <a for=/>URLs</a> are strings and can appear
-anywhere, a specific bidirectional algorithm for rendered <a for=/>URLs</a> would not see wide
-adoption. Bidirectional text interacts with the parts of a <a for=/>URL</a> in ways that can cause
-the rendering to be different from the model. Users of bidirectional languages are thus cautioned
-that this is to be expected, particularly in plain text environments.
+ <p class="note no-backref">Note that non-ASCII characters can be used in <a
+ href="http://unicode.org/faq/idn.html#26">homograph</a> spoofing attacks. Consider detecting <a
+ href="http://www.unicode.org/reports/tr39/#Confusable_Detection">confusable characters</a> or
+ warning when they are in use.
 
-<p>Due to the confusion that can arise between a <a for=/>URL</a>'s <a for=url>host</a>
-and <a for=url>path</a> with bidirectional text, browsers are encouraged to only render a
-<a for=/>URL</a>'s <a for=url>host</a> in places where it is important for users to
-distinguish between the two. E.g., users are expected to make trust decisions based on a
-<a for=/>URL</a>'s <a for=url>host</a> rendered in the address bar.
+ <li><p>URLs are particularly prone to confusion between host and path when they contain
+ bidirectional text, so in this case it is particularly advisable to only render a URL’s <a
+ for=url>host</a> (as recommended in <a>Simplify non-human-readable or irrelevant components</a>).
 
+ <li><p>Bidirectional text should be rendered as if it were in a left-to-right embedding. [[!BIDI]]
 
+ <p class="note no-backref">Unfortunately, as rendered <a for=/>URLs</a> are strings and can appear
+ anywhere, a specific bidirectional algorithm for rendered <a for=/>URLs</a> would not see wide
+ adoption. Bidirectional text interacts with the parts of a <a for=/>URL</a> in ways that can cause
+ the rendering to be different from the model. Users of bidirectional languages are thus cautioned
+ that this is to be expected, particularly in plain text environments.
+
+ <li><p>Other parts of the <a for=/>URL</a> should have their sequences of
+ <a>percent-encoded bytes</a> replaced with code points resulting from
+ <a>percent decoding</a> those sequences converted to bytes, unless that renders those
+ sequences invisible.
```

Comment:
This is primarily about the path right? So maybe we should combine this with the bit two points above whereby we encourage only to render the host?

Reply:
Done